lib/s3-s3.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * AWS.S3-like behavior with failover to a secondary S3 location
+ */
+
+'use strict';
+
+// Used in init() below, this function mimics AWS.S3
+var s3 = function (primaryS3, secondaryS3) {
+  var
+    // request object that proxies to the real AWS.Request object
+    Request = function(realRequest) {
+      var
+        // storage of on() callbacks
+        onFunc = {},
+        // proxying on() callbacks to the real AWS.Request object
+        setupOn = function(realRequest, onType, onFunc) {
+          realRequest.on(onType, function(response) {
+            if (onFunc.hasOwnProperty(onType)) {
+              onFunc[onType](response);
```

Comment:
Yes.  There are other params that don't, but I didn't add them all in given the huge less: http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Request.html.  Other, lesser-used ones, will throw an error.  I will make a note of this in the readme, whenever I get around to that.